torrentool/torrent2.py

```diff
@@ -0,0 +1,444 @@
+        if date is not None:
+            date = datetime.utcfromtimestamp(int(date))
+        return date
+
+    @creation_date.setter
+    def creation_date(self, val):
+        self._struct['creation date'] = timegm(val.timetuple())
+
+    @property
+    def created_by(self):
+        """Optional. Name and version of the program used to create the .torrent"""
+        return self._struct.get('created by')
+
+    @created_by.setter
+    def created_by(self, val):
+        self._struct['created by'] = val
+
+    @property
+    def private(self):
+        """Optional. If True the client MUST publish its presence to get other peers
+        ONLY via the trackers explicitly described in the metainfo file. If False or is not present,
+        the client may obtain peer from other means, e.g. PEX peer exchange, dht.
+        """
+        return self._struct.get('info', {}).get('private', False)
+
+    @private.setter
+    def private(self, val):
+        if not val:
+            try:
+                del self._struct['info']['private']
+            except KeyError:
+                pass
+        else:
+            self._struct['info']['private'] = 1
+
+    @property
+    def name(self):
+        """Torrent's name"""
+        return self._struct.get('info', {}).get('name')
+
+    @name.setter
+    def name(self, val):
+        self._struct['info']['name'] = val
+
+    def to_file(self, filepath=None):
+        """Writes Torrent object into file, either
+
+        :param filepath:
+        """
+        if filepath is None and self._filepath is None:
+            raise TorrentError('Unable to save torrent to file: no filepath supplied.')
+
+        if filepath is None:
+            filepath = '%s.torrent' % self._filepath
+
+        print(filepath)
+        with open(filepath, mode='wb') as f:
+            f.write(self.to_string())
+
+    def to_string(self):
+        """Returns bytes representing torrent file.
+
+        :param str encoding: Encoding used by strings in Torrent object.
+        :rtype: bytearray
+        """
+        return Bencode.encode(self._struct)
+
+    @classmethod
+    def _get_target_files_info(cls, src_path):
+        src_path = u'%s' % src_path  # Force walk() to return unicode names.
+
+        is_dir = isdir(src_path)
+        target_files = []
+
+        if is_dir:
+            for base, _, files in walk(src_path):
+                target_files.extend([join(base, fname) for fname in sorted(files)])
+
+        else:
+            target_files.append(src_path)
+
+        target_files_ = []
+        total_size = 0
+        for fpath in target_files:
+            file_size = getsize(fpath)
+            if not file_size:
+                continue
+            target_files_.append((fpath, file_size, normpath(fpath.replace(src_path, '')).strip(sep).split(sep)))
+            total_size += file_size
+
+        print(target_files_)
+
+        return target_files_, total_size
+
+    def _calc_size(self, size):
```

Comment:
I need to keep what we have now except from the blocksize ill change that to 32kb and add hardcoded limits. torrentool will use this by default unless some of the options is set.